### PR TITLE
Check readme images after building reference and articles

### LINKED ISSUE #2194: False warning for missing (articles) images in README.md
For a README.Rmd with

```r
knitr::include_graphics("vignettes/data_sources.png")
``` 

a first run of `build_site()` leads to 

```r
Warning message:
Missing images in 'README.md': 'articles/data_sources.png'
ℹ pkgdown can only use images in 'man/figures' and 'vignettes' 
```

The image is then present. It is very close to #1980 so should the fix be an extension of https://github.com/r-lib/pkgdown/commit/8b2f4d0a500c911255ce51b19ff193582b4e67bb to vignettes/ images? 

cc @mcguinlu

### PR DESCRIPTION
**Pull request by hadley**
We can't know if the images in a readme exist until we have built the rest of the site. We worked around this a bit for images in `man/figures` but that strategy doesn't work for articles since we actually need to build them. So now we just check in a new step at the end of `build_site()`.

Fixes #2194

### CODE REVIEW

File: R/check.R [hunk 1]
```diff
@@ -23,3 +23,30 @@ check_pkgdown <- function(pkg = ".") {
...
+  pkg <- as_pkgdown(pkg)
+
+  cli::cli_rule("Checking for problems")
+  index_path <- path_index(pkg)
+  if (!is.null(index_path)) {
+    check_missing_images(pkg, index_path, "index.html")
+  }
+}
+
+check_missing_images <- function(pkg, src_path, dst_path) {
```
Comment (maelle) [inline R/check.R:37]: why not check all relative links? thinking of #2216

Comment (maelle) [reply R/check.R:37]: this could be helpful for people trying to add files via assets.

Comment (maelle) [reply R/check.R:37]: maybe using `get_site_paths()` (although that would prevent users from linking to stuff in the dev website, now probably linking to stuff in the dev website isn't a good idea?)

Comment (hadley) [reply R/check.R:37]: Yeah, I think we'll probably accumulate more checks in here, now that we have this function.

**Comment by maelle** [comment]
Different from https://github.com/r-lib/covr/issues/482?

**Comment by hadley** [comment]
@maelle yeah, because the test actually fails when run by covr.

**Pull request merged**